qlib/Swagger.qm

```diff
@@ -0,0 +1,3666 @@
+    MimeTypeMultipartFormData: {},
+};
+
+#! modules available for data serialization and/or deserialization
+public const SerializationModules = map $1.module, MimeDataTypes.iterator(), $1;
+
+#! MIME types for data serialization
+public const MimeContentTypes = keys MimeDataTypes;
+
+#! Valid transfer protocol schemes.
+public const ValidSchemes = ("http", "https", "ws", "wss");
+public const ValidSchemesHash = map {$1: True}, ValidSchemes;
+
+#! Base class for the %Swagger specification objects, wrapping the vendor extensions.
+class ObjectBase {
+    public {
+        #! Allows extensions to the %Swagger Schema.
+        /**
+            The field name MUST begin with \c x-, for example, \c x-internal-id.
+            The value can be \c null, a primitive, an array or an object.
+         */
+        hash vendorExtensions;
+    }
+
+    #! Constructor.
+    public constructor() {
+    }
+
+    #! Constructor.
+    /**
+        @param oh deserialized hash from the source schema description describing the object
+     */
+    public constructor(hash oh) {
+        initialize(oh);
+    }
+
+    #! copy constructor
+    constructor(ObjectBase other) {
+        vendorExtensions = other.vendorExtensions;
+    }
+
+    #! Initialize.
+    /**
+        @param oh deserialized hash from the source schema description
+     */
+    public initialize(hash oh) {
+        vendorExtensions = map {$1.key: $1.value}, oh.pairIterator(), $1.key =~ /^x-/;
+    }
+}
+
+#! Base used by @ref OtherParameter, @ref HeaderObject and @ref SchemaObject.
+class SchemaBase {
+    public {
+        #! See https://tools.ietf.org/html/draft-fge-json-schema-validation-00#section-5.1.2.
+        *float maximum;
+
+        #! See https://tools.ietf.org/html/draft-fge-json-schema-validation-00#section-5.1.3.
+        *float minimum;
+
+        #! See https://tools.ietf.org/html/draft-fge-json-schema-validation-00#section-5.1.2.
+        *bool exclusiveMax;
+
+        #! See https://tools.ietf.org/html/draft-fge-json-schema-validation-00#section-5.1.3.
+        *bool exclusiveMin;
+
+        #! See https://tools.ietf.org/html/draft-fge-json-schema-validation-00#section-5.2.1.
+        *int maxLength;
+
+        #! See https://tools.ietf.org/html/draft-fge-json-schema-validation-00#section-5.2.2.
+        *int minLength;
+
+        #! See https://tools.ietf.org/html/draft-fge-json-schema-validation-00#section-5.2.3.
+        *string pattern;
+
+        #! See https://tools.ietf.org/html/draft-fge-json-schema-validation-00#section-5.3.2.
+        *int maxItems;
+
+        #! See https://tools.ietf.org/html/draft-fge-json-schema-validation-00#section-5.3.3.
+        *int minItems;
+
+        #! See https://tools.ietf.org/html/draft-fge-json-schema-validation-00#section-5.3.4.
+        *bool uniqueItems;
+
+        #! See https://tools.ietf.org/html/draft-fge-json-schema-validation-00#section-5.5.1.
+        hash<string, bool> enum;
```

Comment:
This is wrong because according to documentation, the enum values can have any type, not just `string`.

Reply:
And as seen in `getExampleValue()`, not just `int`, `number` or `boolean` (`float` is not a swagger type).

Reply:
updated implementation with example values and implementation notes to document that a hash is used for quick lookups and that we only support enum values that are convertible to a string for practical reasons